File: Leetcode/labuladong_template/js-template/leetcode/editor/common/listNode.js
export class ListNode {
    constructor(val, next) {
        this.val = (val === undefined ? 0 : val);
        this.next = (next === undefined ? null : next);
    }

    static print(head) {
        let current = head;
        let s = ""
        while (current !== null) {
            s += current.val + (current.next ? "->" : "");
            current = current.next;
        }
        console.log(s);
    }

    static createHead(nums) {
        let head = new ListNode();
        let current = head;
        for (let i = 0; i < nums.length; i++) {
            current.next = new ListNode(nums[i]);
            current = current.next;
        }
        return head.next;
    }
}

File: Leetcode/labuladong_template/java-template/src/main/java/leetcode/editor/common/ListNode.java
package leetcode.editor.common;

public class ListNode {
    public int val;
    public ListNode next;

    public ListNode(int x) {
        val = x;
    }

    public ListNode(int val, ListNode next) {
        this.val = val;
        this.next = next;
    }

    public static ListNode createHead(int[] nums) {
        if (nums == null || nums.length == 0) {
            return null;
        }
        ListNode head = new ListNode(nums[0]);
        ListNode cur = head;
        for (int i = 1; i < nums.length; i++) {
            cur.next = new ListNode(nums[i]);
            cur = cur.next;
        }
        return head;
    }

    public static void print(ListNode head) {
        for (ListNode cur = head; cur != null; cur = cur.next) {
            String arrow = cur.next != null ? "->" : "";
            System.out.print(cur.val + arrow);
        }
    }
}
